### PR TITLE
Fix recompilation-every-update bug in jax_training.py

### PR DESCRIPTION
**Pull request by kuds**
- [x] Understand the issue: `step_fn` returns normalized obs; `update_running_stats` incorrectly receives normalized obs
- [x] Fix `step_fn` to return raw `states.obs` instead of normalized `obs`
- [x] Update training loop to normalize raw obs after updating `obs_stats`
- [x] Code review and security check (no issues found)

### CODE REVIEW

File: environments/shared/jax_training.py [hunk 1]
```diff
@@ -127,9 +126,9 @@ def step_fn(carry, _):
 
             return (new_states, rng), (obs, action, log_prob, value, rewards, dones)
 
-        return step_fn
+        return jax.lax.scan(step_fn, (states, rng), None, length=rollout_len)
```
[the remark this answers is not in the document]
Comment (kuds) [reply environments/shared/jax_training.py:132]: @copilot apply changes based on [this feedback](https://github.com/kuds/mesozoic-labs/pull/368#discussion_r3009816485)
